File: ueueo-framework/ueueo-background-jobs/src/main/java/com/ueueo/backgroundjobs/JsonBackgroundJobSerializer.cs
using System;
using Volo.Abp.DependencyInjection;
using Volo.Abp.Json;

namespace Volo.Abp.BackgroundJobs;

public class JsonBackgroundJobSerializer : IBackgroundJobSerializer, ITransientDependency
{
    private readonly IJsonSerializer _jsonSerializer;

    public JsonBackgroundJobSerializer(IJsonSerializer jsonSerializer)
    {
        _jsonSerializer = jsonSerializer;
    }

    public String Serialize(Object obj)
    {
        return _jsonSerializer.Serialize(obj);
    }

    public Object Deserialize(String value, Type type)
    {
        return _jsonSerializer.Deserialize(type, value);
    }

    public T Deserialize<T>(String value)
    {
        return _jsonSerializer.Deserialize<T>(value);
    }
}

File: ueueo-framework/ueueo-background-jobs/src/main/java/com/ueueo/backgroundjobs/JsonBackgroundJobSerializer.java
package com.ueueo.backgroundjobs;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;

public class JsonBackgroundJobSerializer implements IBackgroundJobSerializer {
    @Override
    public String serialize(Object obj) {
        return JSON.toJSONString(obj);
    }

    @Override
    public <T> T deserialize(String value, Class<T> type) {
        return JSONObject.parseObject(value, type);
    }

}
